rsyslog/undefined_field/undefined_field.go

```diff
@@ -0,0 +1,392 @@
+/**
+ * undefined_field.go -- rsyslog plugin to process undefined fields and do some more treatments [1]
+ *
+ * Interface
+ * Input  - mmexternal plugins receives $! property tree as well as local and global variables in the pipeline.
+ * Output - only $! property tree can be returned to rsyslogd via standard output.
+ *
+ * Due to the spec of rsyslog, outputs from plugins are accumulated to the $! property tree.
+ * To support move undefined fields in the tree and deleting empty fields, undefined_field
+ * sets the processed property tree to $!openshift_logging_all and the result is supposed
+ * to be reset to $! if needed.  See the following config example, where $! is cleaned up
+ * before resetting the result from undefined_field.  If no changes are made in undefined_
+ * field, it returns an empty json.  In that case, rsyslogd does not touch $!.
+ *
+ * How to use undefined_field
+ * 1) compile - go build undefined_field.go
+ * 2) rsyslog config example to call undefined_field
+ *    module(load="mmexternal")
+ *    action(type="mmexternal" binary="/usr/local/bin/undefined_field" interface.input="fulljson")
+ *    if (strlen($!openshift_logging_all) > 0) then {
+ *        set $.openshift_logging_all = $!openshift_logging_all;
+ *          unset $!;
+ *          set $! = $.openshift_logging_all;
```

Comment:
so `unset $!;` then `set $!` works?  I thought it wasn't working?  Or was it `reset $! = $.openshift_logging_all;` that wasn't working?

Reply:
> so `unset $!;` then `set $!` works? I thought it wasn't working? 

It works...  I thought the same - "not working", but maybe there was some other bugs that confused me...

> Or was it `reset $! = $.openshift_logging_all;` that wasn't working?

I believe so.  but let me double check it.

Reply:
> >     Or was it reset $! = $.openshift_logging_all; that wasn't working?
> I believe so. but let me double check it.

I double checked `reset $! = $.openshift_logging_all; ` adds $.openshift_logging_all on top of $! which I don't want to do.